src/Features/Core/Portable/EmbeddedLanguages/DateAndTime/EmbeddedCompletionContext.cs

```diff
@@ -33,31 +52,91 @@ internal partial class DateAndTimeEmbeddedCompletionProvider
                 }
 
                 _replacementSpan = TextSpan.FromBounds(startPosition, context.Position);
+
+                (_userFormatPrefix, _userFormatSuffix) = GetUserFormatParts(virtualChars, startPosition, context.Position);
             }
 
-            private static DateTime s_exampleDateTime = DateTime.Parse("2009-06-15T13:45:30.1234567Z");
-            private static CultureInfo s_enUsCulture = CultureInfo.GetCultureInfo("en-US");
-            private static CultureInfo s_primaryCulture = CultureInfo.CurrentCulture;
-            private static CultureInfo? s_secondaryCulture = s_primaryCulture.Equals(s_enUsCulture) ? null : s_enUsCulture;
+            private static (string prefix, string suffix) GetUserFormatParts(
+                VirtualCharSequence virtualChars, int startPosition, int endPosition)
+            {
+                virtualChars = virtualChars.IsDefault ? VirtualCharSequence.Empty : virtualChars;
```

Comment:
I admit this line looks like voodoo magic because clearly Default and Empty are two concepts.

Reply:
it's got the same semantics as ImmutableArray (and is doc'ed as such) :